src/jit/morph.cpp

```diff
@@ -2301,13 +2300,15 @@ void fgArgInfo::EvalArgsToTemps()
             // Create a placeholder node to put in its place in gtCallLateArgs.
 
             // For a struct type we also need to record the class handle of the arg.
-            CORINFO_CLASS_HANDLE clsHnd = NULL;
+            CORINFO_CLASS_HANDLE clsHnd = NO_CLASS_HANDLE;
 
 #if defined(_TARGET_AMD64_) && !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
 
+            // All structs are either passed (and retyped) as integral types, OR they
+            // are passed by reference.
             noway_assert(argx->gtType != TYP_STRUCT);
 
-#else // !(defined(_TARGET_AMD64_) && !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING))
+#else // !(defined(_TARGET_AMD64_) || defined(FEATURE_UNIX_AMD64_STRUCT_PASSING))
```

Comment:
I agree with what you intended to do.

However, In general, I'm somewhat ambivalent about tagging #else and #endif of these (marginally) complicated conditions, because, well, it's complicated...